### PR TITLE
add-pundit

### PR DESCRIPTION
**Pull request by Ritumka**

### CODE REVIEW

**Review by tluu5 — COMMENTED** [review]
great job adding the Pundit gem and the user_not_authorized method

**Review by tluu5 — COMMENTED** [review]
Added if statement on user#show

**Review by tluu5 — APPROVED** [review]
You are ready.

**Comment by armstrca** [comment]
In the real world, branches are rarely created off of other branches. They're usually just branched directly off of main and intended to merge back directly to main.

In this context, it kind of makes sense to branch the way you've done here, because different people are reviewing both PRs and this one is dependent on the content of the earlier PR. So if the earlier one hasn't merged to main yet, I don't have to sort through as many changes that are already getting looked at in the other PR by Mary  (`mp-add-pundit` has more changes compared to `main` than compared to `mp-add-authorization`).

That all said, it would be best practice to merge the previous PR to main first, and then change this PR so that it merges into `main` rather than `mp-add-authorization` (that's also what the official assignment requirement is).

File: app/controllers/photos_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
   before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
+  # before_action :ensure_user_is_authorized, only: [:show]
```
Comment (armstrca) [inline app/controllers/photos_controller.rb]: Removed commented-out code

**Comment by Ritumka** [comment]
> In the real world, branches are rarely created off of other branches. They're usually just branched directly off of main and intended to merge back directly to main.
> 
> In this context, it kind of makes sense to branch the way you've done here, because different people are reviewing both PRs and this one is dependent on the content of the earlier PR. So if the earlier one hasn't merged to main yet, I don't have to sort through as many changes that are already getting looked at in the other PR by Mary (`mp-add-pundit` has more changes compared to `main` than compared to `mp-add-authorization`).
> 
> That all said, it would be best practice to merge the previous PR to main first, and then change this PR so that it merges into `main` rather than `mp-add-authorization` (that's also what the official assignment requirement is).

This PR has only commits related to Pundit assignment and it compares `mp-add-pundit` branch with `mp-add-authorization` (not `main`). But I got you! Thank you.

File: app/controllers/photos_controller.rb (hunk 1 above)
Comment (Ritumka) [reply app/controllers/photos_controller.rb]: Done!

**Review by armstrca — CHANGES_REQUESTED** [review]
Overall looks great!
The assignment does technically require this to be a merge to main, so just change the commit to `main` from `mp-add-pundit` and resolve any merge conflicts that might come up.

**Review by armstrca — APPROVED** [review]

**Pull request merged**